README.rst

```diff
@@ -721,6 +721,8 @@ The following directives are recognized:
 | * ``default``: ``proto_library``, ``go_proto_library``, and ``go_library``                 |
 |   rules are generated using ``@io_bazel_rules_go//proto:def.bzl``. Only one                |
 |   of each rule may be generated per directory. This is the default mode.                   |
+| * ``file``: a ``proto_library`` rule is generated for every .proto file.                   |
+|   .proto files can be grouped using ``proto_group`` option.                                |
```

Comment:
This doesn't really make sense to me. If `file` mode puts one `.proto` file in each `proto_library` target, it doesn't seem meaningful to group them at all. Why would `proto_group` have any effect?

Reply:
It was mostly a side-effect of the way I implemented the per-file option, but I think it's pretty useful. The `proto_group` option allows you to override the default name of the `proto_library` rule. If for some reason someone wanted to logically group proto files, but still default to single proto file libraries, this allows that.

You could also use this option to rename a `proto_library`, if that was desired for whatever reason.

I think I could improve my wording here. How about:

> The `proto_library` a .proto file ends up in can be overridden by using the `proto_group` option.

Reply:
> The proto_group option allows you to override the default name of the proto_library rule.

I wonder if a better way to do that in this mode would be to match on `srcs` when finding targets to merge? In language/proto/kinds.go, it looks like `proto_library` will already do that. That seems a much more intuitive workflow: users can just change the name of the target in the build file, and Gazelle keeps using the new name thereafter. `proto_group` seems much less intuitive to me for this use case.

> If for some reason someone wanted to logically group proto files, but still default to single proto file libraries, this allows that.

Gazelle's probably a poor fit for that level of customization. I'd recommend manually maintaining build files for directories like this.

In any case, if `proto_group` does this, it needs to be tested to make sure it doesn't break later on.

Reply:
Thanks for bringing this up. After doing some more digging I found that the `proto_group` method didn't even work in file mode :facepalm: (else if vs if). Untested code is broken code... Anyway, I dropped the docs about it and the (faulty) code that would have made it work, as suggested.

I was unaware of `MatchAttrs`, that should handle most use cases that I was thinking about.

Reply:
@jayconrod have you had a chance to look at this?

Reply:
Regrettably, I no longer have bandwidth to work on Gazelle. I'll add @bazelbuild/go-maintainers in case anyone else can take a look.

Reply:
@achew22 are you able to take a quick look at this?